src/opnsense/scripts/unbound/start.sh

```diff
@@ -26,11 +26,26 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
-set -e
-
 # prepare and startup unbound, so we can easily background it
 
-chroot -u unbound -g unbound / /usr/local/sbin/unbound-anchor -a /var/unbound/root.key
+# if the root.key file is missing or damaged, run unbound-anchor
+/usr/local/sbin/unbound-checkconf /var/unbound/unbound.conf 2> /dev/null 
+
+if [ $? -eq 1 ]; then
+	# unbound-anchor has undefined behaviour if file is corrupted, start clean 
+	rm -f /var/unbound/root.key
+
+	# if we are in forwarding mode, prefer to use the configured system nameservers
+	if [ $(/usr/local/sbin/pluginctl -g unbound.forwarding) ]; then
```

Comment:
that should read

```suggestion
	if [ -n "$(/usr/local/sbin/pluginctl -g unbound.forwarding)" ]; then
```

but honestly I don't think we can trust unbound.forwarding as it may be set but empty historically. don't we have a new forwarding knob as of forwarding MVC restructure?

Reply:
> don't we have a new forwarding knob as of forwarding MVC restructure?
We don't. Query forwarding here refers to the legacy forwarding option as listed in General, using the system nameservers - forwarding as part of MVC restructure doesn't touch this.
 
> but honestly I don't think we can trust unbound.forwarding as it may be set but empty historically
any suggestions? I have no idea how this was registered in old versions.

Reply:
for pluginctl this doesn't work if it was stored as `<forwarding/>` ... I have a similar issue with dns server registering at the moment and worked around it by moving logic from shell scripts  to php backend altogether.

we cound write a /var/unbound/resolv.conf.root file if we have that situation and just include it if it exists? I'm not 100% sure trusting /etc/resolv.conf is good because it may have other servers in there not used by unbound forwarding in particular.

Reply:
Are we certain this was the case historically? Current legacy doesn't do this - though it is better to assume it happens I suppose. 

As far as I'm aware the only servers populating /etc/resolv.conf are manually assigned nameservers in system->settigns->general, as well as dynamically added DHCP DNS servers. This falls within expectations for forwarding purposes (but only for query forwarding in unbound->general). Honestly, I'd really like to move this legacy section to MVC so every nameserver can be included here as well.